packages/web3-utils/src/index.js

```diff
@@ -240,31 +241,32 @@ export const toWei = (number, unit) => {
  *
  * @method toChecksumAddress
  *
- * @param {String} address the given HEX address
+ * @param {string} address the given HEX address
  *
- * @returns {String}
+ * @param {number} chain where checksummed address should be valid.
+ *
+ * @returns {string} address with checksum applied.
  */
-export const toChecksumAddress = (address) => {
-    if (typeof address === 'undefined') return '';
+export const toChecksumAddress = (address, chainId = null) => {
+    if (typeof address !== 'string') {
+        return '';
+    }
 
     if (!/^(0x)?[0-9a-f]{40}$/i.test(address))
         throw new Error(`Given address "${address}" is not a valid Ethereum address.`);
 
-    address = address.toLowerCase().replace(/^0x/i, '');
-    const addressHash = utils.keccak256(address).replace(/^0x/i, '');
+    const stripAddress = stripHexPrefix(address).toLowerCase();
+    const prefix = chainId != null ? (chainId.toString() + '0x') : '';
+    const keccakHash = Hash.keccak256(prefix + stripAddress).toString('hex').replace(/^0x/i, '');
     let checksumAddress = '0x';
 
-    for (let i = 0; i < address.length; i++) {
-        // If ith character is 9 to f then make it uppercase
-        if (parseInt(addressHash[i], 16) > 7) {
-            checksumAddress += address[i].toUpperCase();
-        } else {
-            checksumAddress += address[i];
-        }
-    }
+    for (let i = 0; i < stripAddress.length; i++)
+        checksumAddress += parseInt(keccakHash[i], 16) >= 8 ?
```

Comment:
Could you change this line to: 
`` let output = parseInt(keccakHash[i], 16) >= 8 ?
            stripAddress[i].toUpperCase() :
            stripAddress[i].toLowerCase();
``